jamventProto/jamctrllib/JamCtrlMgrInterface.h

```diff
@@ -0,0 +1,181 @@
+
+        /// Tidal volume
+        /// Units: Milliliters
+        /// Range: 10 to 1500
+        IntegerNumType tidalVol_ml;
+
+        /// Respiratory rate
+        /// Units: None. Breaths per minute
+        /// Range: 1 to 40
+        IntegerNumType respRate;
+
+        /// Inhale/Exhale ratio.
+        /// Units: None. Choice of ratios
+        /// Range: 0 (1:4) to 6 4:1)
+        ///   0 = 1:4
+        ///   1 = 1:3
+        ///   2 = 1:2
+        ///   3 = 1:1
+        ///   4 = 2:1
+        ///   5 = 3:1
+        ///   6 = 4:1
+        IntegerNumType ieRatio;
+
+        /// Positive end expiratory pressure
+        /// Units: Bar
+        /// Range: 0.0 to 0.0196133 Bar (internal units), 0.0 to 20.0 cmH2O (display units)
+        FloatNumType peep_bar;
+
+        // Valve states.
+        bool valveAO2Open; ///< True when valve A is open. False when closed.
+        bool valveBAirOpen; ///< True when valve B is open. False when closed.
+        bool valveCInhaleOpen; ///< True when valve C is open. False when closed.
+        bool valveDExhaleOpen; ///< True when valve D is open. False when closed.
+
+        // Graph data
+        /// Reservior pressure (high)
+        /// Units: Bar
+        /// Range: 0.0 Bar to 5.0 Bar
+        FloatNumType pRes_bar; // Pressure in reservior
+
+        /// System pressure (low)
+        /// Units: Bar
+        /// Range: 0.01 Bar to 1.2 Bar
+        FloatNumType pSys_bar; // Pressure at the patient
+
+        /// O2 percentage in mix
+        /// Units: Percentage (%)
+        /// Range: 21.0 to 100.0
+        FloatNumType o2_pc;
+
+        // @todo Fill in details.  Just a placeholder for now
+        /// Lung volume
+        FloatNumType lungVol_ml;
+
+        DataIn() : fiO2_pc{0}, tidalVol_ml{0}, respRate{0},
+                   ieRatio{0}, peep_bar{0.0}, valveAO2Open{false}, valveBAirOpen{false},
+                   valveCInhaleOpen{false}, valveDExhaleOpen{false}, pRes_bar{0.0},
+                   pSys_bar{0.0}, o2_pc{0.0}, lungVol_ml{0.0} {};
+    };
+
+    JamCtrlMgrInterface() {};
+    virtual ~JamCtrlMgrInterface() {};
+
+    // Data from the system
+    /**
+    * @brief Get the latest data from the system.
+    * @return Data from the system
+    */
+    virtual const DataIn getCtrlData() = 0;
+
+    // Data out to the system.
+    /**
+    * @brief Set valve A open or closed
+    * @param open True to open the valve. False to close.
+    */
+    virtual void setValveAopen(bool open) = 0;
+
+    /**
+    * @brief Set valve B open or closed
+    * @param open True to open the valve. False to close.
+    */
+    virtual void setValveBopen(bool open) = 0;
+
+    /**
+    * @brief Set valve C open or closed
+    * @param open True to open the valve. False to close.
+    */
+    virtual void setValveCopen(bool open) = 0;
+
+    /**
+    * @brief Set valve D open or closed
+    * @param open True to open the valve. False to close.
+    */
+    virtual void setValveDopen(bool open) = 0;
```

Comment:
Should we rename these functions similarly to how we discussed naming the valves themselves by including oxy/air/inhale/exhale details in the name?

Reply:
Yes.  Just missed this.  I will update.